rpcs3/rpcs3qt/debugger_frame.cpp

```diff
@@ -532,93 +532,65 @@ void debugger_frame::WritePanels()
 void debugger_frame::ShowGotoAddressDialog()
 {
 	QDialog* diag = new QDialog(this);
-	diag->setWindowTitle(tr("Enter expression"));
+	diag->setWindowTitle(tr("Go To Address"));
 	diag->setModal(true);
 
 	// Panels
 	QVBoxLayout* vbox_panel(new QVBoxLayout());
-	QHBoxLayout* hbox_address_preview_panel(new QHBoxLayout());
 	QHBoxLayout* hbox_expression_input_panel = new QHBoxLayout();
 	QHBoxLayout* hbox_button_panel(new QHBoxLayout());
 
-	// Address preview
-	QLabel* address_preview_label(new QLabel(diag));
-	address_preview_label->setFont(m_mono);
-
 	// Address expression input
 	QLineEdit* expression_input(new QLineEdit(diag));
 	expression_input->setFont(m_mono);
 	expression_input->setMaxLength(18);
-	expression_input->setFixedWidth(190);
 
+	if (auto thread = cpu.lock(); !thread || thread->id_type() != 2)
+	{
+		expression_input->setValidator(new QRegExpValidator(QRegExp("^(0[xX])?0*[a-fA-F0-9]{0,8}$")));
+	}
+	else
+	{
+		expression_input->setValidator(new QRegExpValidator(QRegExp("^(0[xX])?0*[a-fA-F0-9]{0,5}$")));
```

Comment:
Why do these have '0*'? I think this would be best left off, since it's basically included in the next block. As is, you could match silly strings like `0x00000000000000000000000000000000000000000000000000000000000000000000000000000000f`, which is not good. I don't know if the other Qt settings prevent this tho.

Reply:
input max length is 18. its just the regexp.